scripts/uninstall.sh

```diff
@@ -43,30 +43,24 @@ if [ ! -f $FUNCTION_ACTIVATION_CONFIG ]; then
     echo
 fi
 
+readonly GCP_PROJECT=$("$YQ" e '.googleCloud.required.gcpProjectId' $FUNCTION_ACTIVATION_CONFIG)
 readonly GCP_SERVICE_ACCOUNT=$("$YQ" e '.googleCloud.common.serviceAccount' $FUNCTION_ACTIVATION_CONFIG)
 readonly GCP_PUBSUB_TOPIC=$("$YQ" e  '.googleCloud.metrics.pubSubTopic' $FUNCTION_ACTIVATION_CONFIG)
 readonly GCP_FUNCTION_NAME=$("$YQ" e '.googleCloud.metrics.function' $FUNCTION_ACTIVATION_CONFIG)
 readonly GCP_SCHEDULER_NAME=$("$YQ" e '.googleCloud.metrics.scheduler' $FUNCTION_ACTIVATION_CONFIG)
 readonly DYNATRACE_URL_SECRET_NAME=$("$YQ" e '.googleCloud.common.dynatraceUrlSecretName' $FUNCTION_ACTIVATION_CONFIG)
 readonly DYNATRACE_ACCESS_KEY_SECRET_NAME=$("$YQ" e '.googleCloud.common.dynatraceAccessKeySecretName' $FUNCTION_ACTIVATION_CONFIG)
 readonly SELF_MONITORING_DASHBOARD_NAME="dynatrace-gcp-function Self monitoring"
+readonly GCP_FUNCTION_REGION=$("$YQ" e '.googleCloud.required.cloudFunctionRegion' $FUNCTION_ACTIVATION_CONFIG)
 
 GCP_ACCOUNT=$(gcloud config get-value account)
 echo -e "You are now logged in as [$GCP_ACCOUNT]"
-echo
-DEFAULT_PROJECT=$(gcloud config get-value project)
-
-echo "Please provide the GCP project from which monitoring function should be removed. Default value: [$DEFAULT_PROJECT] (current project)"
-while ! [[ "${GCP_PROJECT}" =~ ^[a-z]{1}[a-z0-9-]{5,29}$ ]]; do
-    read -p "Enter GCP project ID: " -i $DEFAULT_PROJECT -e GCP_PROJECT
-done
-echo ""
 
 echo "- set current project to [$GCP_PROJECT]"
 gcloud config set project $GCP_PROJECT
 
 echo "Discovering instances to remove"
-REMOVE_FUNCTION=$(gcloud functions list --filter=name:$GCP_FUNCTION_NAME --format="value(name)")
+REMOVE_FUNCTION=$(gcloud functions list --filter=name:$GCP_FUNCTION_NAME --format="value(name)" --regions=$GCP_FUNCTION_REGION)
```

Comment:
All other resources (PubSub topic, Scheduler job, etc.) are global and we don't need this region in order to delete them, right?

Reply:
right